zebra-chain/src/transaction/txidhash.rs

```diff
@@ -0,0 +1,54 @@
+//! Transaction ID hashing. Contains code for generating the Transaction ID
+//! from the transaction, using hashing.
+use std::{convert::TryInto, io};
+
+use super::Transaction;
+
+use crate::serialization::{sha256d, ZcashSerialize};
+
+use super::Hash;
```

Comment:
Nit: Maybe it makes sense to join these imports in a single block?

```suggestion
use super::{Hash, Transaction};
use crate::serialization::{sha256d, ZcashSerialize};
```